static/js/biomedical/protein/protein_protein_interaction_graph.tsx

```diff
@@ -63,6 +73,7 @@ export class ProteinProteinInteractionGraph extends React.Component<
     this.state = {
       depth: DEFAULTS.DEPTH,
       graphData: null,
+      loading: true,
```

Comment:
when using loading, we usually do:

- set it to false in the constructor
- set it to true right before data fetch is called (so probably first line in the data fetch function)
- set to to false when the promise completes (in the `.then()` and the `.catch()`)

Reply:
Ah ok thanks for letting me know! I think probably I wanna come back to loading in a future PR then when I add the loading spinner.  For the update button behavior I can think of two possibilities then:
1. We rename the current `loading` to `toUpdate` (so later on we would have both a `loading` flag and a `toUpdate` flag)
2. We create 3 new properties in the state for each of the text inputs, and the update button sets the original properties `onClick`:
```
this.state.depth  = this.state.depthInput;
this.state.numInteractions = this.state.numInteractionsInput;
this.state.depth = this.state.scoreThresholdInput;
```

What do you think?

Reply:
Ah, I see what you're trying to do now. Yes, let's come back to the real loading in a future PR. For here, I think we could:

- on button clicked, call fetchData()
- in this didComponentUpdate function, just only fetch data when `!_.isEqual(prevProps, this.props)`

Reply:
wow that's clean!! thanks I'll do this

Reply:
Oh I realized we can also do `fetchData` in `componentDidMount` and remove it completely from `componentDidUpdate`.

Reply:
we want to keep it in componentDidUpdate in case the proteinDcid changes but the component is never unmounted (although I'm not sure it'll happen in this case, but good to have it there still)

Reply:
Ah good point will do!

Reply:
also, I don't think you need the first check (return if there is no change in props and state) in the componentDidUpdate function anymore

Reply:
oh! yes another nice catch!